tests/e2e/unified_test.go

```diff
@@ -153,6 +157,78 @@ func testFixturesInSeries(ctx context.Context, t *testing.T, testPause bool, can
 					createPausedCC(ctx, t, h.GetClient())
 				}
 
+				if os.Getenv("E2E_GCP_TARGET") == "vcr" {
+					// Stop recording after tests finish and write to cassette
+					t.Cleanup(func() {
+						err := h.VCRRecorder.Stop()
+						if err != nil {
+							t.Errorf("[VCR] Failed stop vcr recorder: %v", err)
+						}
+					})
+
+					hook := func(i *cassette.Interaction) error {
+						var requestHeadersToRemove = []string{
+							"Authorization",
+							"User-Agent",
+						}
+						for _, header := range requestHeadersToRemove {
+							delete(i.Request.Headers, header)
+						}
+
+						var responseHeadersToRemove = []string{
+							"Cache-Control",
+							"Server",
+							"Vary",
+							"X-Content-Type-Options",
+							"X-Frame-Options",
+							"X-Xss-Protection",
+							"Date",
+							"Etag",
+						}
+						for _, header := range responseHeadersToRemove {
+							delete(i.Response.Headers, header)
+						}
+
+						replaceFunc := func(s string) string {
+							result := strings.Replace(s, uniqueID, "uniqueid111111", -1)
+							result = strings.Replace(result, project.ProjectID, "cnrm-user", -1)
+							return result
+						}
+
+						i.Request.Body = replaceFunc(i.Request.Body)
+						i.Response.Body = replaceFunc(i.Response.Body)
+						i.Request.URL = replaceFunc(i.Request.URL)
+
+						s := i.Response.Body
+						obj := make(map[string]any)
+						if err := json.Unmarshal([]byte(s), &obj); err != nil {
+							klog.Fatalf("error from json.Unmarshal(%q): %v", s, err)
+						}
+						toReplace, _, _ := unstructured.NestedString(obj, "user")
+						if len(toReplace) != 0 {
+							s = strings.Replace(s, toReplace, "user@google.com", -1)
+						}
+
+						// A very hacky way to replace actual debug info from log.
```

Comment:
Nit: Would be great if the hacky way is explained (why hacky, any future TODO, etc).

Reply:
Actually I plan to remove this code block in PR #1393 and re-generate the log file. 
To be able to speed up test, I'd like to remove unnecessary interactions from the recorder log. For example, failed HTTP logs with error code 404/400/403 and repeated logs come from the retry transport.